File: binary tree/levelOrderTraversal.cpp
// Level Order Traversal - Hackerrank

void levelOrder(Node *root)
{
    queue<Node *> q;
    q.push(root);
    while (q.empty() == false)
    {
        Node *node = q.front();
        cout << node->data << " ";
        q.pop();

        // Enqueuing left and right children of current node to queue
        if (node->left != NULL)
        {
            q.push(node->left);
        }

        if (node->right != NULL)
        {
            q.push(node->right);
        }
    }
}

File: binary tree/levelOrderTraversal.java
// Leetcode - 102 - Binary Tree Level Order Traversal
public List<List<Integer>> levelOrder(TreeNode root) {
    List<List<Integer>> result = new ArrayList<>();
    if(root == null) {
        return result;
    }
    
    Queue<TreeNode> queue = new LinkedList<>();
    queue.add(root);
    while(!queue.isEmpty()) {
        int size = queue.size();
        List<Integer> currentLevel = new ArrayList<>();
        for(int i = 0; i < size; i++) {
            TreeNode current = queue.remove();
            currentLevel.add(current.val);
            if(current.left != null) {
                queue.add(current.left);
            }
            if(current.right != null) {
                queue.add(current.right);
            }
        }
        result.add(currentLevel);
    }
    return result;
}